src/Runner.Common/JobServerQueue.cs

```diff
@@ -292,14 +317,62 @@ private async Task ProcessWebConsoleLinesQueueAsync(bool runOnce = false)
                         {
                             try
                             {
-                                // we will not requeue failed batch, since the web console lines are time sensitive.
-                                if (batch[0].LineNumber.HasValue)
+                                if (this._websocketConnectTask != null)
                                 {
-                                    await _jobServer.AppendTimelineRecordFeedAsync(_scopeIdentifier, _hubName, _planId, _jobTimelineId, _jobTimelineRecordId, stepRecordId, batch.Select(logLine => logLine.Line).ToList(), batch[0].LineNumber.Value, default(CancellationToken));
+                                    // lazily await here, we are already in the background task here
+                                    await this._websocketConnectTask;
                                 }
-                                else
+
+                                var pushedLinesViaWebsocket = false;
+                                if (this._websocketClient != null)
+                                {
+                                    var linesWrapper =  batch[0].LineNumber.HasValue? new TimelineRecordFeedLinesWrapper(stepRecordId, batch.Select(logLine => logLine.Line).ToList(), batch[0].LineNumber.Value):
+                                        new TimelineRecordFeedLinesWrapper(stepRecordId, batch.Select(logLine => logLine.Line).ToList());
+                                    var jsonData = StringUtil.ConvertToJson(linesWrapper);
+                                    try
+                                    {
+                                        totalBatchedLinesAttemptedByWebsocket++;
+                                        // It should be okay to wait for the result since we are already doing this in the background and doing it one by one
+                                        await this._websocketClient.SendAsync(new ArraySegment<byte>(Encoding.UTF8.GetBytes(jsonData)), WebSocketMessageType.Text, true, default(CancellationToken));
```

Comment:
I think we might have to break this into 1024 bytes each and then finally send `true` as the last message